query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/fields/LdapAttributeName.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.ldap.fields;
+
+import static org.codice.ddf.admin.ldap.commons.LdapMessages.invalidUserAttribute;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.codice.ddf.admin.api.report.ErrorMessage;
+import org.codice.ddf.admin.common.fields.base.scalar.StringField;
+
+public class LdapAttributeName extends StringField {
+
+    public static final String DEFAULT_FIELD_NAME = "attriName";
+
+    public static final String FIELD_TYPE_NAME = "LdapAttributeName";
+
+    public static final String DESCRIPTION =
+            "The name of an attribute that holds information about an LDAP entry.";
+
+    public LdapAttributeName() {
+        super(DEFAULT_FIELD_NAME, FIELD_TYPE_NAME, DESCRIPTION);
+    }
+
+    public LdapAttributeName(String fieldName) {
+        super(fieldName, FIELD_TYPE_NAME, DESCRIPTION);
+    }
+
+    @Override
+    public List<ErrorMessage> validate() {
+        List<ErrorMessage> errors = super.validate();
+        if (!errors.isEmpty()) {
+            return errors;
+        }
+
+        errors.addAll(validate(this));
+        return errors;
+    }
+
+    // TODO: 7/7/17 - tbatie - This validate should be reformated once there is a generic way to create MapField objects that contain different value field.
+    public static List<ErrorMessage> validate(StringField attribute) {
+        List<ErrorMessage> errors = new ArrayList<>();
+
+        if (attribute.getValue() != null && !attribute.getValue()
```

Comment:
✏️ I don't normally suggest burning up space for temporary data, but this would read a lot better like this:
```java
String attributeName = attribute.getValue();
if (attributeName == null) {
    return errors;
}

if (attributeName.matches([first regex]) || attributeName.matches([second regex])) {
    return errors;
}

errors.add(invalidUserAttribute(attribute.path()));
return errors;